frame/support/src/traits/tokens/nonfungibles_v2.rs

```diff
@@ -0,0 +1,243 @@
+
+	/// Provide the appropriate witness data needed to destroy an item.
+	fn get_destroy_witness(collection: &Self::CollectionId) -> Option<Self::DestroyWitness>;
+
+	/// Destroy an existing fungible item.
+	/// * `collection`: The `CollectionId` to be destroyed.
+	/// * `witness`: Any witness data that needs to be provided to complete the operation
+	///   successfully.
+	/// * `maybe_check_owner`: An optional account id that can be used to authorize the destroy
+	///   command. If not provided, we will not do any authorization checks before destroying the
+	///   item.
+	///
+	/// If successful, this function will return the actual witness data from the destroyed item.
+	/// This may be different than the witness data provided, and can be used to refund weight.
+	fn destroy(
+		collection: Self::CollectionId,
+		witness: Self::DestroyWitness,
+		maybe_check_owner: Option<AccountId>,
+	) -> Result<Self::DestroyWitness, DispatchError>;
+}
+
+/// Trait for providing an interface for multiple collections of NFT-like items which may be
+/// minted, burned and/or have attributes set on them.
+pub trait Mutate<AccountId, ItemConfig>: Inspect<AccountId> {
+	/// Mint some `item` of `collection` to be owned by `who`.
+	///
+	/// By default, this is not a supported operation.
+	fn mint_into(
+		_collection: &Self::CollectionId,
+		_item: &Self::ItemId,
+		_who: &AccountId,
+		_config: &ItemConfig,
+	) -> DispatchResult {
+		Err(TokenError::Unsupported.into())
+	}
+
+	/// Burn some `item` of `collection`.
+	///
+	/// By default, this is not a supported operation.
+	fn burn(
+		_collection: &Self::CollectionId,
+		_item: &Self::ItemId,
+		_maybe_check_owner: Option<&AccountId>,
+	) -> DispatchResult {
+		Err(TokenError::Unsupported.into())
+	}
+
+	/// Set attribute `value` of `item` of `collection`'s `key`.
+	///
+	/// By default, this is not a supported operation.
+	fn set_attribute(
+		_collection: &Self::CollectionId,
+		_item: &Self::ItemId,
+		_key: &[u8],
+		_value: &[u8],
+	) -> DispatchResult {
+		Err(TokenError::Unsupported.into())
+	}
+
+	/// Attempt to set the strongly-typed attribute `value` of `item` of `collection`'s `key`.
+	///
+	/// By default this just attempts to use `set_attribute`.
+	fn set_typed_attribute<K: Encode, V: Encode>(
+		collection: &Self::CollectionId,
+		item: &Self::ItemId,
+		key: &K,
+		value: &V,
+	) -> DispatchResult {
+		key.using_encoded(|k| value.using_encoded(|v| Self::set_attribute(collection, item, k, v)))
+	}
+
+	/// Set attribute `value` of `collection`'s `key`.
+	///
+	/// By default, this is not a supported operation.
+	fn set_collection_attribute(
+		_collection: &Self::CollectionId,
+		_key: &[u8],
+		_value: &[u8],
+	) -> DispatchResult {
+		Err(TokenError::Unsupported.into())
+	}
+
+	/// Attempt to set the strongly-typed attribute `value` of `collection`'s `key`.
+	///
+	/// By default this just attempts to use `set_attribute`.
+	fn set_typed_collection_attribute<K: Encode, V: Encode>(
```

Comment:
These traits probably need more a in-depth review than I have time to do rn.

Reply:
Since I only copy+pasted them from v1, let's work on them in a separate PR, wdyt?